contracts/zaps/BaseZap.sol

```diff
@@ -0,0 +1,40 @@
+// SPDX-License-Identifier: AGPL-3.0
+pragma solidity 0.6.12;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import {
+    SafeERC20,
+    IERC20
+} from "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
+import "../../../interfaces/yearn/IYVault.sol";
+
+abstract contract BaseZap is Ownable {
+    using SafeERC20 for IERC20;
+
+    IERC20 public want;
+    IYVault public yVault;
+
+    function updateVaultAddress(address _vault) external onlyOwner {
+        yVault = IYVault(_vault);
+        want.safeApprove(_vault, uint256(-1));
+    }
+
+    function zapEthIn() external payable virtual {
+        require(false, "not impented");
+    }
+
+    function zapEthIn(uint256 slippageAllowance) external payable virtual {
```

Comment:
i think we can add two methods `zapInAll` and `zapIn` with signature `zapInAll(address token, uint256 amount, uint256 slippage)`

Reply:
I think maybe we should have two types of zaps
- Eth zaps
- ERC20 zaps

Reply:
may be difficult to scale per vault, each vault = 2 zap. where N is number of vaults N ^2 unless we have a way to centralize all zaps for all vaults in less contracts but that sounds complicated.

Supporting two entry points in the same zap allows less tracking for UI also, one button for eth for them is the same contract from the other button for token.